crates/zizmor/src/audit/bot_conditions.rs

```diff
@@ -114,6 +134,195 @@ impl Audit for BotConditions {
 }
 
 impl BotConditions {
+    /// Check if the workflow has events where bot conditions are relevant.
+    fn has_relevant_events(workflow: &Workflow) -> bool {
+        use github_actions_models::workflow::Trigger;
+
+        match &workflow.on {
+            Trigger::BareEvent(event) => Self::is_relevant_event(event),
+            Trigger::BareEvents(event_list) => event_list
+                .iter()
+                .any(Self::is_relevant_event),
+            Trigger::Events(event_map) => {
+                !matches!(event_map.issue_comment, OptionalBody::Missing)
```

Comment:
Out of curiosity, why are we filtering out the `OptionalBody::Missing` case here? I don't think that affects the overall event payload in this context.

Reply:
AFAIK, the `OptionalBody::Missing` properly checks if any event is present as a trigger (either with or without configuration), which determines whether bot conditions are relevant for workflows that have these events as triggers. I might have misunderstood this comment:

https://github.com/zizmorcore/zizmor/blob/ec2f674ca324f191307c5d5421a448ca65811a5c/crates/github-actions-models/src/workflow/event.rs#L156-L159

Reply:
No worries!